firefox-ios/Client/Frontend/Browser/Tabs/Middleware/TabManagerMiddleware.swift

```diff
@@ -436,6 +439,14 @@ class TabManagerMiddleware {
                                                          actionType: TabPanelMiddlewareActionType.refreshInactiveTabs)
             store.dispatch(refreshAction)
 
+            // Refresh the active tabs panel. Can only happen if the user is in normal browsering mode (not private).
+            // Related: FXIOS-10010, FXIOS-9954, FXIOS-9999
+            let model = getTabsDisplayModel(for: false, shouldScrollToTab: false, uuid: uuid)
+            let refreshActiveTabsPanelAction = TabPanelMiddlewareAction(tabDisplayModel: model,
+                                                                        windowUUID: uuid,
+                                                                        actionType: TabPanelMiddlewareActionType.refreshTabs)
+            store.dispatch(refreshActiveTabsPanelAction)
+
```

Comment:
https://github.com/user-attachments/assets/896ff876-7344-4d3f-bfba-f7fb1fd6cffe

Not sure if this is just a weird thing with the debug menu 10s timeout for tab activity being enabled but it looks like this doesn't work anymore?

Reply:
Really great catch @dataports, I've updated my logic and added additional tests! 🎉 

I _think_ all the edge cases have now been accounted for.


https://github.com/user-attachments/assets/0c7db232-66c2-486b-9f77-babb82289d13

Reply:
(P.S. still hold off on reviewing, one other little thing I'll tweak tomorrow)